jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostClassifierSuite.scala

```diff
@@ -308,4 +309,42 @@ class XGBoostClassifierSuite extends FunSuite with PerTest {
     val xgb = new XGBoostClassifier(paramMap)
     xgb.fit(repartitioned)
   }
+
+  test("feature size of predication data should be same with model's") {
+    val modelPath = getClass.getResource("/model/0.82/model").getPath
+    val model = XGBoostClassificationModel.read.load(modelPath)
+    val r = new Random(0)
+    val df = ss.createDataFrame(Seq.fill(100)(r.nextInt(2)).map(i => (i, i))).
+      toDF("feature", "label")
+    val assembler = new VectorAssembler()
+      .setInputCols(df.columns.filter(!_.contains("label")))
+      .setOutputCol("features")
+    intercept[Exception] {
+      model.transform(assembler.transform(df)).show()
+    }
+  }
+
+  test("feature size must be same for distributed train") {
```

Comment:
ditto

Reply:
and also you may want to have both neg and pos test

i.e. some test to capture exception (and you should not capture a general exception, maybe runtimeexception, etc.

Reply:
Done, same with previous comments. the test case was copied from https://github.com/dmlc/xgboost/blob/master/jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostClassifierSuite.scala#L291 and it's a positive one.

Reply:
> and also you may want to have both neg and pos test
> 
> i.e. some test to capture exception (and you should not capture a general exception, maybe runtimeexception, etc.

Yes, in some cases, SparkException was captured and I don't know why. I added the error msg checking